types/node/events.d.ts

```diff
@@ -755,7 +804,10 @@ declare module "events" {
                  * Returns a reference to the `EventEmitter`, so that calls can be chained.
                  * @since v0.1.26
                  */
-                removeAllListeners(eventName?: Key<unknown, T>): this;
+                /* eslint-disable @definitelytyped/no-unnecessary-generics */
+                removeAllListeners<EventName extends EventNames<Events>>(eventName: EventName): this;
+                removeAllListeners<EventName extends string | symbol>(eventName?: EventName): this;
+                /* eslint-enable @definitelytyped/no-unnecessary-generics */
```

Comment:
Most of the multi-generic-overload signatures have been leaned down, but there are still a couple remaining (here and `emit`). Can these use `EventNameParam` as well?

As before, these overloaded methods will currently raise a compiler error if called on unions of emitters (#71059).

Reply:
Pretty sure I put tests in for all of this, and if not I did at least test it because I was worried about that. I'll add test cases for this if they're missing (and fix if I did just miss these)